### PR TITLE
wiki: added Rest API docs

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #1099 

Added wiki page about rest api

### CODE REVIEW

**Comment by olenagerasimova** [comment]
@dgarus please, review

File: .wiki/Rest-api.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Artipie management Rest API
+
+Artipie provides Rest API to manage [repositories](./Configuration-Repository), [users](./Configuration-Credentials) 
+and [storages aliases](./Configuration-Storage#Storage-Aliases). API is self-documented with [Swagger](https://swagger.io/)
+interface, Swagger documentation pages are available on URLs `http://localhost:8081/api/index-{layout}.html` where
```
Comment (dgarus) [inline .wiki/Rest-api.md:5]: `http://localhost:8081/api/index-{layout}.html` -> `http://{host}:{api-port}/api/index-{layout}.html`

File: .wiki/Rest-api.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+`layout` is the layout you run Artipie with, `org` or `flat`.
+
+In Swagger documentation have three definitions - Repositories, Users and Auth Token. You can switch
+between the definitions with the help of "Select a definition" listbox.
+
+<img src="https://user-images.githubusercontent.com/14931449/193015387-3e25f937-7f23-4b27-884c-f183ca9dc8a0.png" alt="Swagger documentation" width="400"/>
+
+All Rest API endpoints require JWT authentication token to be passed in `Authentification` header. 
+The token can be issued with the help of `POST /api/v1/oauth/token` request on "Auth Token"
```
Comment (dgarus) [inline .wiki/Rest-api.md:14]: on **the** "Auth Token" definition page

File: .wiki/Rest-api.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+`layout` is the layout you run Artipie with, `org` or `flat`.
+
+In Swagger documentation have three definitions - Repositories, Users and Auth Token. You can switch
+between the definitions with the help of "Select a definition" listbox.
+
+<img src="https://user-images.githubusercontent.com/14931449/193015387-3e25f937-7f23-4b27-884c-f183ca9dc8a0.png" alt="Swagger documentation" width="400"/>
+
+All Rest API endpoints require JWT authentication token to be passed in `Authentification` header. 
+The token can be issued with the help of `POST /api/v1/oauth/token` request on "Auth Token" 
+definition page in Swagger. Once token is received, copy it, open another definition, press
```
Comment (dgarus) [inline .wiki/Rest-api.md:15]: Is there a similar description in the Swagger documentation?
I think about a link on this doc.

File: .wiki/Rest-api.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+Rest API allows to manage repository settings: read, create, update and remove operations are supported. 
+Note, that JSONs, accepted by Rest endpoints, are equivalents of the YAML repository settings. Which means, 
+that API accepts all the repository specific settings fields which are applicable to the repository. 
+Choose repository you are interested in from [this table](./Configuration-Repository#Supported-repository-types) 
+to learn all the details. 
+
+Rest API provides method to rename repository `PUT /api/v1/{repo_name}/move` (`{repo_name}` is the 
+name of the repository, includes username in the case of `org` layout) and move all the data
+from repository with the `{repo_name}` to repository with new name (new name is provided in json
```
Comment (dgarus) [inline .wiki/Rest-api.md:28]: json -> JSON
or small symbols, but everywhere should be the same style

File: .wiki/Rest-api.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+Rest API allows to manage repository settings: read, create, update and remove operations are supported. 
+Note, that JSONs, accepted by Rest endpoints, are equivalents of the YAML repository settings. Which means, 
+that API accepts all the repository specific settings fields which are applicable to the repository. 
+Choose repository you are interested in from [this table](./Configuration-Repository#Supported-repository-types) 
+to learn all the details. 
+
+Rest API provides method to rename repository `PUT /api/v1/{repo_name}/move` (`{repo_name}` is the 
+name of the repository, includes username in the case of `org` layout) and move all the data
+from repository with the `{repo_name}` to repository with new name (new name is provided in json 
+request body, check Swagger docs to learn the format). Response is returned immoderately, but data
```
Comment (dgarus) [inline .wiki/Rest-api.md:29]: immoderately ? Is it typo?

File: .wiki/Rest-api.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+Rest API provides method to rename repository `PUT /api/v1/{repo_name}/move` (`{repo_name}` is the 
+name of the repository, includes username in the case of `org` layout) and move all the data
+from repository with the `{repo_name}` to repository with new name (new name is provided in json 
+request body, check Swagger docs to learn the format). Response is returned immoderately, but data 
+manipulation is performed in asynchronous mode, so to make sure data transfer is complete, 
+call `HEAD /api/v1/{repo_name}` and verify status `404 NOT FOUND` is returned.
+
+## Storage aliases
+[Storage aliases](./Configuration-Storage#Storage-Aliases) can also be managed with Rest API, 
+there are methods to read, create, update and remove aliases. Note, that storage aliases can be
```
Comment (dgarus) [inline .wiki/Rest-api.md:35]: > Note, that storage aliases can be set on different levels: for repository, user or common level.

It looks like a description for the [Storage aliases](./Configuration-Storage#Storage-Aliases) section.
We should provide examples of configurations for different levels.

Comment (dgarus) [inline .wiki/Rest-api.md:35]: > Note, that storage aliases can be set on different levels: for repository, user or common level.

It looks like a description for the [Storage aliases](./Configuration-Storage#Storage-Aliases) section.
We should provide examples of configurations for different levels.

File: .wiki/Rest-api.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+level, can be used only for this repository, such configuration can allow to easily switch 
+storages (and, for example, configure data replication for these storages). Storage aliases for 
+user are available for all repositories of the user and common aliases can be used in 
+any Artipie repository.
+
+## Users management API
+
+Use Rest API to obtain list of the users, check user info, add, update or remove user. Also, it's
+possible to change password by calling `POST /api/v1/{username}/alter/password` method providing
+old and new password in json request body.
```
Comment (dgarus) [inline .wiki/Rest-api.md:46]: json -> JSON

**Review by dgarus — CHANGES_REQUESTED** [review]
Thanks!
Please look at my comments.

File: .wiki/Rest-api.md (hunk 6 above)
Comment (olenagerasimova) [reply .wiki/Rest-api.md:35]: Configurations are absolutely the same, only URLs differs

File: .wiki/Rest-api.md (hunk 7 above)
Comment (olenagerasimova) [reply .wiki/Rest-api.md:46]: I have not found JSON (capitalized) in our md files, so why change here?

File: .wiki/Rest-api.md (hunk 4 above)
Comment (olenagerasimova) [reply .wiki/Rest-api.md:28]: I have not found JSON (capitalized) in our md files, so why change here?

File: .wiki/Rest-api.md (hunk 3 above)
Comment (olenagerasimova) [reply .wiki/Rest-api.md:15]: I do not think there is such documentation in swagger

**Comment by olenagerasimova** [comment]
@dgarus thanks, corrected and answered

File: .wiki/Rest-api.md (hunk 4 above)
Comment (dgarus) [reply .wiki/Rest-api.md:28]: ok, small symbols are fine

**Review by dgarus — APPROVED** [review]

**Pull request merged**